src/command.ts

```diff
@@ -27,22 +30,55 @@ export class Command {
             },
         };
 
-        const exitCode = await exec.exec(executable, args, finalExecOptions);
+        if (execOptions.group) {
+            const groupName = [ executable, ...args ].join(" ");
+            core.startGroup(groupName);
+        }
+
+        try {
+            const exitCode = await exec.exec(executable, args, finalExecOptions);
 
-        if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
-            // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
-            // which saves some clicking when debugging.
-            let error = `${path.basename(executable)} exited with code ${exitCode}`;
-            if (stderr) {
-                error += `\n${stderr}`;
+            if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
+                // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
+                // which saves some clicking when debugging.
+                let error = `${path.basename(executable)} exited with code ${exitCode}`;
+                if (stderr) {
+                    error += `\n${stderr}`;
+                }
+                throw new Error(error);
             }
-            throw new Error(error);
+
+            return {
+                exitCode, output: stdout, error: stderr,
+            };
         }
 
-        return {
-            exitCode,
-            stdout,
-            stderr,
-        };
+        finally {
+            if (execOptions.group) {
+                core.endGroup();
+            }
+        }
+    }
+
+    public static async tag(image: string, tags: string[]): Promise<void> {
+        // get docker cli
+        let dockerPath: string;
+
+        try {
+            dockerPath = await io.which("docker", true);
+        }
+        catch (error) {
+            core.debug(error);
+            core.setFailed(
```

Comment:
should this be a `throw` instead of a `setFailed` so that the top-level function exits?